lib/src/resources/movie_api_provider.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:async';
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import '../models/movie.dart';
+import '../../utils/constants.dart';
+import 'movie_strings.dart';
+
+class MovieApiProvider {
+  http.Client client = http.Client();
+
+  Future<MovieModel> fetchAllMovies() async {
+    final _response = await client.get(
+      Uri.parse(
+        MovieStrings.uriClient,
+      ),
+    );
+    return resp(_response);
+  }
+
+  Future<MovieModel> searchByMovieName(String nameMovie) async {
+    final _response = await client.get(
+      Uri.parse(
+        MovieStrings.uriSearchClient + nameMovie,
+      ),
+    );
+    return resp(_response);
+  }
+
+  MovieModel resp(http.Response _response) {
```

Comment:
"resp" is not a descriptive name, avoid cropping words as much as posible when naming elements

Reply:
done